src/gh/diffCheck/diffCheck/df_vizualization.py

```diff
@@ -0,0 +1,293 @@
+        trans = rg.Transform.PlaneToPlane(rg.Plane.WorldXY, plane)
+        for geo in legend_geometry:
+            geo.Transform(trans)
+
+    return legend_geometry
+
+
+def create_histogram(values, min_value, max_value, steps=100,
+                     plane=rg.Plane.WorldXY, total_height=10,
+                     scaling_factor=0.01, spacing=0):
+    """
+    Create a histogram in Rhino with a polyline representing value frequencies.
+    """
+
+    height = total_height/steps
+
+    histogram_geometry = []
+
+    # Calculate the size of each bin
+    bin_size = (max_value - min_value) / steps
+
+    # Initialize the frequency counts for each bin
+    frequencies = [0] * (steps + 1)
+
+    # if values is nested list, flatten it
+    if isinstance(values[0], list):
+        values = [item for sublist in values for item in sublist]
+
+    # Count the frequencies of values in each bin
+    for value in values:
+        if value < min_value:
+            value = min_value
+        elif value > max_value:
+            value = max_value
+        bin_index = (value - min_value) // bin_size
+        bin_index = int(bin_index)
+        frequencies[bin_index] += 1
+
+    # Create points for the polyline representing the histogram
+    points = []
+    for i in range(steps+1):
+
+        bar_height = frequencies[i] * scaling_factor
+        points.append(rg.Point3d(- bar_height - (1.5 * height), i * (spacing + height), 0))
+
+    # Create the polyline and add it to the histogram geometry
+    polyline = rg.Curve.CreateInterpolatedCurve(points, 1)
+    histogram_geometry.append(polyline)
+
+    if plane != rg.Plane.WorldXY:
+        trans = rg.Transform.PlaneToPlane(rg.Plane.WorldXY, plane)
+        for geo in histogram_geometry:
+            geo.Transform(trans)
+
+    return histogram_geometry
+
+
+def filter_values_based_on_valuetype(results, settings):
+
+    if settings.valueType == "Dist":
+
+        min_value = min(min(sublist) for sublist in results.distances)
+        max_value = max(max(sublist) for sublist in results.distances)
+        values = results.distances
+
+    elif settings.valueType == "MSE":
+
+        values = results.distances_mse
+        min_value = min(values)
+        max_value = max(values)
+
+    elif settings.valueType == "MAX":
+
+        values = results.distances_max_deviation
+        min_value = min(values)
+        max_value = max(values)
+
+    elif settings.valueType == "MIN":
+        values = results.distances_min_deviation
+        min_value = min(values)
+        max_value = max(values)
+
+    elif settings.valueType == "STD":
+
+        values = results.distances_sd_deviation
+        min_value = min(values)
+        max_value = max(values)
+
+    # threshold values
+    if settings.lower_threshold is not None:
+        min_value = settings.lower_threshold
+    if settings.upper_threshold is not None:
+        max_value = settings.upper_threshold
+
+    return values, min_value, max_value
```

Comment:
shouldn't this be a class method of the `DFVizResults` object. Imo it has little to do in visualization.. It is also a bit of an empty object so it would be nice to add to it this method I think.

Reply:
yes!!! Thanks for the suggestion :)